src/scratch/index.js

```diff
@@ -42,6 +42,40 @@ export const scratchWorkspaceInit = async (scratch_area_name, scratch_div_name)
         
         Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(main_xml), workspace);
 
+        /* temperaily download & upload button */
+        document.getElementById('download').onclick = () => {
+            const downloadContent = Blockly.Xml.domToText(
+                Blockly.Xml.workspaceToDom(workspace)
+            );
+            
+            const download = (filename, text) => {
+                const element = document.createElement('a');
+                element.setAttribute('href', `data:text/plain;charset=utf-8,${  encodeURIComponent(text)}`);
+                element.setAttribute('download', filename);
+              
+                element.style.display = 'none';
+                document.body.appendChild(element);
+              
+                element.click();
+              
+                document.body.removeChild(element);
+            };
+
+            download('binary-bot.xml', downloadContent);
```

Comment:
We don't use the `download` or `downloadContent` function elsewhere, do we need two separate functions for them? We could just execute the logic right away.